app/models/forms/attribute_reset_collection.rb

```diff
@@ -0,0 +1,23 @@
+module Forms
+  class AttributeResetCollection
+    ResetData = Struct.new(:attributes_to_reset, :master_attribute, :enabled_value)
+
+    def initialize
+      @collection = []
+    end
+
+    def add(*args)
+      @collection << ResetData.new(*args)
+    end
+
+    def perform(form)
+      @collection.each do |reset_obj|
+        next if form.public_send(reset_obj.master_attribute) == reset_obj.enabled_value
+
+        reset_obj.attributes_to_reset.each do |attribute|
+          form.public_send("#{attribute}=", nil)
```

Comment:
We could have it look at the schema with something like this:

```ruby
form.model.column_defaults[attribute]
```

Applied to the reset collection thing it might look something like this:

```ruby
class AttributeResetCollection
    ...
    def perform(form)
      @collection.each do |reset_obj|
        next if form.public_send(reset_obj.master_attribute) == reset_obj.enabled_value

        reset_obj.attributes_to_reset.each do |attribute|
          reset_value = form.model.column_defaults[attribute]
          form.public_send("#{attribute}=", reset_value)
        end
      end
    end
end
```